sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -88,20 +91,35 @@ case class CreateTableLikeCommand(
       sourceTableDesc.provider
     }
 
+    val newStorage = if (fileFormat.inputFormat.isDefined) {
+      sourceTableDesc.storage.copy(
```

Comment:
It may be more clear to show what we retained from the source table here:
```
fileFormat.copy(
  compressed = sourceTableDesc.storage.compressed,
  properties = sourceTableDesc.storage.properties
)
```

Reply:
and then it comes the question: if user specifies a different serde in `CREATE TABLE LIKE`, does it make sense to retain the compressed flag and serder properties from the source table?

Reply:
In current spark implementation, we don't have any opportunity to set `compressed=true`. So I think it's ok to just use the source's compressed property.
For `fileFormat.properties`, it's a little tricky, could wen use `sourceTableDesc.storage.properties ++ properties`?
How about this?
Before:
CatalogTable.properties always uses empty
CatalogTable.storage.properties always uses source table's storage.properties
After:
CatalogTable.properties always uses TBLPROPERTIES if user set, or empty.
CatalogTable.storage.properties always uses source table's storage.properties ++ TBLPROPERTIES if user set.

Reply:
Does hive allow users to specify serde properties in CREATE TABLE LIKE?

Reply:
@cloud-fan No. hive doesn't. But I am not sure it's a bug or on purpose. Here is the testing in Hive:
```shell
hive> create table t1 (key int) ROW FORMAT SERDE 'org.apache.hadoop.hive.serde2.OpenCSVSerde' WITH SERDEPROPERTIES ('a'='a') STORED AS TEXTFILE;;
OK
Time taken: 10.595 seconds

hive> desc formatted t1;
OK
# col_name            	data_type           	comment
key                 	string              	from deserializer

# Detailed Table Information
...

# Storage Information
SerDe Library:      	org.apache.hadoop.hive.serde2.OpenCSVSerde
...
Storage Desc Params:
	a                   	a
Time taken: 0.152 seconds, Fetched: 29 row(s)

hive> create table t2 like t1 ROW FORMAT SERDE 'org.apache.hadoop.hive.serde2.OpenCSVSerde' WITH SERDEPROPERTIES ('b'='b');
OK
Time taken: 0.048 seconds
hive> desc formatted t2;
OK
# col_name            	data_type           	comment
key                 	string              	from deserializer

# Detailed Table Information
...

# Storage Information
SerDe Library:      	org.apache.hadoop.hive.serde2.OpenCSVSerde
...

Storage Desc Params:
	a                   	a
Time taken: 0.062 seconds, Fetched: 31 row(s)
```

Reply:
So maybe current PR is good enough to get Hive compatibility:
Before:
CatalogTable.properties always uses empty
CatalogTable.storage.properties always uses source table's storage.properties
After:
CatalogTable.properties always uses TBLPROPERTIES if user set, or empty.
CatalogTable.storage.properties always uses source table's storage.properties

Reply:
I will address this first.
```scala
fileFormat.copy(
  compressed = sourceTableDesc.storage.compressed,
  properties = sourceTableDesc.storage.properties
)
```

Reply:
serder properties are dedicated to the serde. I don't think it's reasonable to retain the serde properties from source table when we change the serde.

I think we should simply use the newly specified `CatalogStorageFormat`.

Reply:
Just confirmation:
Before:
CatalogTable.properties always uses empty
CatalogTable.storage.properties always uses source table's storage.properties
After:
CatalogTable.properties always uses TBLPROPERTIES if user set, or empty.
CatalogTable.storage.properties always uses the properties from we specified `CatalogStoragrFormat`

Reply:
yes

Reply:
So we need to add `ROW FORMAT` in CTL for Hive table only.
Below three cases will create a Hive table:
1. provider set to hive
2. provider is empty but fileFormat is used
3. provider and fileFormat are all unused, source table is hive table

Reply:
> yes

A little late, can we retain the source `CatalogTable.properties` ? And use new properties to merge it.
Here is a case:
1. a hive table that use parquet and set tblproperties `parquet.compression`.
2. do spark `create table like`.
3. new table do not have the properties.